lib/eventbus.js

```diff
@@ -0,0 +1,159 @@
+'use strict';
+
+const _        = require('lodash');
+const P        = require('bluebird');
+const bunyan          = require('bunyan');
+
+const EventInvalidError = require('../lib/validator').EventInvalidError;
+
+
+/**
+ * Represents and Eventbus process status result.
+ * Each event passed to Eventbus process() will map
+ * to an final EventStatus.  If the status is a failure
+ * this.context should contain the error.
+ */
+class EventStatus {
+    constructor(status, context, event) {
+        this.status = status;
+        this.context = context;
+        this.event = event;
+    }
+
+    /**
+     * static 'enum' constant for possible status types.
+     */
+    static STATUSES() {
```

Comment:
Why's this a function?

Reply:
Ye, but you could just `ValidatorCache.STATUSES = []`

Reply:
Oh ya! done.